components/VM_Arm/docs/vm_memory_configuration.md

```diff
@@ -0,0 +1,83 @@
+<!--
+     Copyright 2023, DornerWorks
+     SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# VM Memory Configuration
+
+This document overviews how the seL4 VMM provides guests with the a block of memory.
+
+## Basic Configuration
+
+In the platform specific `devices.camkes` file, there are macros that define the VM's RAM region:
+
+```
+#define VM_RAM_BASE 0x40000000
+#define VM_RAM_SIZE 0x20000000
+
+vm0.vm_address_config = {
+    "ram_base" : VAR_STRINGIZE(VM_RAM_BASE),
+    "ram_paddr_base" : VAR_STRINGIZE(VM_RAM_BASE),
+    "ram_size" : VAR_STRINGIZE(VM_RAM_SIZE),
+};
+
+vm0.simple_untyped24_pool = 33;
+
+```
+
+This configuration lets VM0 see 512MB at 0x40000000. When running as a hypervisor, seL4 uses stage 2
+mappings to map an Intermediate Physical Address (IPA) to a Physical Address (PA). This means that
+0x40000000 is the IPA that the guest sees, and the associated PA could be any available page in the
+system. The pages backing the guest mappings are provided by the `simple_untyped24_pool`.
```

Comment:
A note here would be good, why there have to be 33 and not just 32 memory blocks in the pool.

Reply:
Done.